docgen/parser/parser.go

```diff
@@ -155,14 +155,63 @@ func parseValues(path string, comments map[string]*Comment, values []*google_pro
 	return res
 }
 
+// extractMapFields generates an array of fields that correspond to map type fields for messages, which are stored in
+// the NestedType field of the DescriptorProto struct
+func extractMapFields(message *google_protobuf.DescriptorProto, path string, comments map[string]*Comment) []*Field {
+	fields := make([]*Field, len(message.NestedType))
+	for i, n := range message.GetNestedType() {
+		// a nested type of map will always have two fields for key and value
+		if len(n.GetField()) != 2 {
+			continue
+		}
+		nestedTypeName := strings.TrimSuffix(n.GetName(), "Entry")
```

Comment:
I find this fragile, like the other comment about `Entry`. There's probably a way to do this that doesn't require messing with strings manually, like comparing type pointers.

Reply:
I added a check for MapEntry type above so this logic should hold as long as it is an entry